src/lib/storage/storage_manager.cpp

```diff
@@ -10,38 +10,59 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
+  _tables.insert({name, table});
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
+  size_t elements_erased = _tables.erase(name);
+
+  if (elements_erased != 1) {
```

Comment:
For understandability, I would also make this a `< 1` (it might not be instantaneously clear that `elements_erased` is either 0 or 1).